### PR TITLE
Define D1B, D4 and D20 diffractometers at ILL

### LINKED ISSUE #18452: Add IDFs for D1B, D4 and D20
Define D1B, D4 and D20 diffractometers at ILL.

### PR DESCRIPTION
**Pull request by gvardany**
This PR adds the IDFs for D4C, D20, D1B diffractometers at ILL. 
They are generated with the scripts in that should be merged first.

* [mantidgeometry:PR83](https://github.com/mantidproject/mantidgeometry/pull/83)
* [mantidgeometry:PR84](https://github.com/mantidproject/mantidgeometry/pull/84)
* [mantidgeometry:PR85](https://github.com/mantidproject/mantidgeometry/pull/85)

Also for existing D2B, the L2 distance and monitor position are corrected.

Detailed detector definitions can be found here:
* [D1B](https://github.com/mantidproject/documents/blob/master/Project-Management/ILL/Instrument_Definitions/D1B/D1B.md)
* [D4](https://github.com/mantidproject/documents/blob/master/Project-Management/ILL/Instrument_Definitions/D4/D4.md)
* [D20](https://github.com/mantidproject/documents/blob/master/Project-Management/ILL/Instrument_Definitions/D20/D20.md)

The descriptions are preliminary confirmed by the instrument scientists.

**To test**:

`LoadEmptyInstrument` with the new IDFs, check the instrument view.
Make sure that the detector models match their descriptions, most importantly, pay attention on:

-  **2&theta; spacing** - the difference of thetas between two neighbouring pixels
-  **2&theta; coverage** - the full 2-theta acceptance of the detectors
-  **initial 2&theta;** - the angle of the first detector pixel.

Fixes #18452  .

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [x] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [x] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: instrument/D2B_Definition.xml [hunk 1]
```diff
@@ -14,17 +14,33 @@ name="D2B" valid-from="1900-01-31 23:59:59" valid-to="2100-01-31 23:59:59" last-
       <handedness val="right" />
     </reference-frame>
   </defaults>
-  <component type="moderator">
-    <location z="-2" />
+  <component type="monochromator">
```
Comment (ianbush) [inline instrument/D2B_Definition.xml]: The changes for D2B are not mentioned in the PR. Can the generator scripts be updated, and a similar doc as for the others be created? This could go in a separate PR.

Comment (gvardany) [inline instrument/D2B_Definition.xml]: I have added a note about D2B. Currently there is no script for generating the D2B definition. This and it's description can be added later.

**Review by ianbush — COMMENTED** [review]
Everything looks good, it is consistent with the instrument descriptions linked to in the PR.

The only suggested change I would make is to have the monitors with ID 0.

**Comment by gvardany** [comment]
I have set the monitor IDs to zero. Corresponding change in mantidgeometry is in [mantidgeometry:PR94](https://github.com/mantidproject/mantidgeometry/pull/94)

**Comment by ianbush** [comment]
All looks good, :shipit:.

**Comment by NickDraper** [comment]
resolved confiict in release notes, All looks good, merging
